### PR TITLE
Add variant for hosted-site-migration flow

### LINKED ISSUE #90654: Site Migration: Implement new `site-migration` flow variant
We have a start on this here: ~https://github.com/Automattic/wp-calypso/pull/90180~ https://github.com/Automattic/wp-calypso/pull/90661

This PR needs to be ~~rebased~~, tested, and deployed before we update external references to point to it.

This PR handles the happy path of a user that:

* Arrives to setup/hosted-site-migration
* Sees the destination picker screen and creates a site by selecting Create a new one
* Picks a Free Plan
* Arrives at the instructions screen. At this step, the destination is already transferred to Atomic and the Migrate Guru plugin is installed

### LINKED ISSUE #90655: Site Migration: Convert new flow variant to a signup flow
The new flow variant introduced in #90180 should be designated as a signup flow.

### PR DESCRIPTION
**Pull request by andres-blanco**
Closes https://github.com/Automattic/wp-calypso/issues/90654
Closes https://github.com/Automattic/wp-calypso/issues/90655
Related to https://github.com/Automattic/wp-calypso/issues/90694

~I've added the DO NOT MERGE label because~
* ~I still want to add tests for the new variant path.~
* ~I want https://github.com/Automattic/wp-calypso/pull/90663 to be merged before this to handle the potential conflicts before merging~

## Proposed Changes

TLDR, this PR implements the happy path for navigating into the new `/setup/import-hosted-site` flow.

The idea is to create an alternative to `/setup/import-hosted-site` that uses Migrate Guru as underlaying migration tech. This PR adds a variation of the current site migration flow with these basic additions:

* You can navigate to it directly by going to `/setup/hosted-site-migration`.
* You get to pick an existing site to migrate into or create a new one.
* If your source site is not on Wordpress then you are redirected to the import flow (this is not done yet in this PR)
Once you have a destination site, you are redirected straight to the migrate/import screen.
The flow continues in the same way as the site-migration flow does.

**BIG MISSING PIECES**

* Back button handling. Navigating to `/setup/hosted-site-migration` shows a `< Back` button with undefined behavior.
* Redirection to import flow if the source site is not on Wordpress, this is not working yet, we need to figure out what to do when there's no destination site. We can find what the current import-hosted does and mimic that.
* More exhaustive testing on the variant path.

## Why are these changes being made?

paYKcK-4BA-p2 

## Testing Instructions
This happy path is ready to test:

* Go to `/setup/hosted-site-migration`
* Select create a new site
* When prompted on the `What do you want to Migrate?` screen, select **Everything**
* See the instructions screen, including your Migrate Guru key

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Comment by andres-blanco** [comment]
we should hold on merging this until https://github.com/Automattic/wp-calypso/pull/90663 is merged to avoid extra noise.

**Comment by sixhours** [comment]
I updated the first step of the testing instructions to say go to `/setup/hosted-site-migration` because I think that's the intent; it said `/setup/import-hosted-site` which is the old flow.

**Comment by markbiek** [comment]
Maybe I'm misunderstanding something but I never get the option to pick the Free Plan as it says in the testing instructions.

If I do Import Only (Free), I just go into the old Import flow. If I do everything, I end up on the Upgrade Plan step of the migration flow.

https://github.com/Automattic/wp-calypso/assets/917632/ba1a1fbe-7715-4c65-9675-285d2de7501c

**Comment by andres-blanco** [comment]
> Maybe I'm misunderstanding something but I never get the option to pick the Free Plan as it says in the testing instructions.
> 
> If I do Import Only (Free), I just go into the old Import flow. If I do everything, I end up on the Upgrade Plan step of the migration flow.
> 

Again I've mixed Free trial for Free plan. My idea was to select Free Trial. 😬 . I've updated instructions now

**Review by markbiek — APPROVED** [review]
LGTM!

**Comment by sixhours** [comment]
Visual nit we should follow up on in a new PR, the site picker is cramped: 

Screenshot 2024-05-14 at 7 58 46 AM

I think we need to use the wide or full-width step layout for this.

File: client/landing/stepper/declarative-flow/registered-flows.ts [hunk 1]
```diff
@@ -168,4 +169,16 @@ const siteMigrationFlow: Record< string, () => Promise< { default: Flow } > > =
 	  }
 	: {};
 
-export default { ...availableFlows, ...videoPressTvFlows, ...siteMigrationFlow };
+const hostedSiteMigrationFlow: Record< string, () => Promise< { default: Flow } > > = {
+	[ HOSTED_SITE_MIGRATION_FLOW ]: () =>
+		import(
+			/* webpackChunkName: "hosted-site-migration-flow" */ '../declarative-flow/hosted-site-migration-flow'
```
Comment (sixhours) [inline client/landing/stepper/declarative-flow/registered-flows.ts:175]: Just curious, what does the webpackChunkName comment do?

**Review by sixhours — APPROVED** [review]
LGTM!
